pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -343,31 +343,37 @@ func calcConnectionsHash(
 	clusterNames := con.Clusters
 	var buffer bytes.Buffer
 	for _, c := range clusterNames {
-		clusterObj, _ := observer.GetCluster(ns, c)
+		clusterObj, clusterErr := observer.GetCluster(ns, c)
 		buffer.WriteString(c)
 		var specNum string
-		// extra careful while dereferencing
-		if clusterObj.Status.SpecGenerationToProcess == nil {
-			specNum = "nil"
-		} else {
-			specNum = strconv.Itoa(
-				int(*clusterObj.Status.SpecGenerationToProcess))
+		if clusterErr == nil {
+			// extra careful while dereferencing
+			if clusterObj.Status.SpecGenerationToProcess == nil {
+				specNum = "nil"
+			} else {
+				specNum = strconv.Itoa(
+					int(*clusterObj.Status.SpecGenerationToProcess))
+			}
 		}
 		buffer.WriteString(specNum)
```

Comment:
If clusterErr is non-nil and therefore `string` is "" (not "nil"), does this code do what we want it to?

Reply:
yes I think so, if the cluster is non-nil then specNum would be "" which would generate new Hash and retrigger config meta generator which would reset cluster connections to nil inside pods which is what we want

Reply:
Hmm. FWIW this is different from how the configmap and secret connections are handled below; in those cases if the err is non-nil we don't write anything into the buffer.